src/service-worker/core.js

```diff
@@ -624,6 +624,10 @@ export function handleFetch(request, maybeClientId) {
       // If not, let's fetch and cache the request.
       return fetchJsFile(cache, versionedRequest, pathname, version);
     });
+  }).catch(err => {
+    // Throw error out of band.
+    Promise.reject(err);
```

Comment:
"This throws the error out-of-band with the promise chain." If we didn't do this, the enclosing promise chain would be passed to `event.respondWith`, which installs a `#catch` block, which means it's rejection is handled. We have to create a new rejected promise that is not handled to get the `unhandledrejection` event.